clients/privacy-center/pages/api/fides-js.ts

```diff
@@ -178,31 +201,60 @@ export default async function handler(
       const userLanguageString =
         fidesLocale || req.headers["accept-language"] || DEFAULT_LOCALE;
 
-      fidesDebugger(
+      log.debug(
         `Fetching relevant experiences from server-side (${userLanguageString})...`,
       );
 
-      /*
-       * Since we don't know what the experience will be when the initial call is made,
-       * we supply the minimal request to the api endpoint with the understanding that if
-       * TCF is being returned, we want the minimal version. It will be ignored otherwise.
-       */
-      experience = await fetchExperience({
-        userLocationString: fidesRegionString,
-        userLanguageString,
-        fidesApiUrl: getFidesApiUrl(),
-        propertyId,
-        requestMinimalTCF: true,
-      });
-      fidesDebugger(
-        `Fetched relevant experiences from server-side (${userLanguageString}).`,
-      );
-      experienceIsValid(experience);
+      // Define how we want to handle the scenario when the API fails to gives us an experience.
+      // By default the behavior is to return an empty experience, we can override that with
+      // the `MISSING_EXPERIENCE_BEHAVIOR` setting. This allows us to explicitly throw an
+      // error in certain cases.
+      const missingExperienceHandler =
+        missingExperienceBehaviors[serverSettings.MISSING_EXPERIENCE_BEHAVIOR];
+
+      try {
+        /*
+         * Since we don't know what the experience will be when the initial call is made,
+         * we supply the minimal request to the api endpoint with the understanding that if
+         * TCF is being returned, we want the minimal version. It will be ignored otherwise.
+         */
+        experience = await pRetry(
+          () =>
+            fetchExperience({
+              userLocationString: fidesRegionString,
+              userLanguageString,
+              fidesApiUrl: getFidesApiUrl(),
+              propertyId,
+              requestMinimalTCF: true,
+              missingExperienceHandler,
+            }),
+          {
+            retries: PREFETCH_MAX_RETRIES,
+            factor: PREFETCH_BACKOFF_FACTOR,
+            minTimeout: PREFETCH_RETRY_MIN_TIMEOUT_MS,
+            onFailedAttempt: (error) => {
+              log.debug(
+                `Attempt to get privacy experience failed, ${error.retriesLeft} remain. Error message was: `,
+                error.message,
+              );
+            },
+          },
+        );
+        log.debug(
+          `Fetched relevant experiences from server-side (${userLanguageString}).`,
+        );
+        experienceIsValid(experience);
+      } catch (error) {
+        log.error(error);
```

Comment:
Will this be a good error message? I'm assuming probably not by default (something super specific) so I'd do this:

```suggestion
        log.error("Error fetching experience from server-side Fides API. Error: ", error);
```

Reply:
Done.